ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+   calendar see the event.  If set to "private", then when a sharee
+   fetches the event the server MUST only return properties that are:
+
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 25]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   *  the basic time and metadata properties of the JSCalendar Event
+      object as specified in Section 4.4.3 of [RFC8984]; or
+   *  properties that are wholly derived from these permitted properties
+      (i.e., utcStart, utcEnd); or
+   *  Additional CalendarEvent properties not derived from the
+      JSCalendar Event data (i.e., id, baseEventId, calendarIds,
+      isDraft, isOrigin).
+
+   If "privacy" is set to "secret", the server MUST behave as though the
+   event does not exist for all users other than the principal that owns
+   the calendar.
+
+5.7.  CalendarEvent/changes
+
+   This is a standard "/changes" method as described in Section 5.2 of
+   [RFC8620].
+
+   Synthetic ids generated by the server expanding recurrences in
+   "CalendarEvent/query" do not appear in "CalendarEvent/changes"
+   responses; only the ids of events as actually stored on the server.
+
+5.8.  CalendarEvent/set
+
+   This is a standard "/set" method as described in Section 5.3 of
+   [RFC8620], with the following extra argument:
+
+   *  *sendSchedulingMessages*: Boolean (default: false)
+      If true then any changes to scheduled events will be sent to all
+      the participants (if the server is the origin of the event) or
+      back to the origin (otherwise).  If false, the changes only affect
+      this account and no scheduling messages will be sent.
+      The server may send the scheduling message via any of the methods
+      defined on the sendTo property of a participant (if the server is
+      the origin) or the replyTo property of the event (otherwise) that
+      it supports.  If no supported methods are available, the server
+      MUST reject the change with a noSupportedScheduleMethods SetError.
+      At time of writing, the most common interoperable protocol for
+      sending scheduling methods between participants on different
+      servers is iMIP [RFC5546].  A future document will provide
+      recommendations of what iMIP messages to send based on the change
+      for best compatibility.
+
+   An id may represent a server-expanded single instance of a recurring
+   event if the client asked the server to expand recurrences in
+   "CalendarEvent/query".  When the synthetic id for such an instance is
+   given, the server MUST process an update as an update to the
+   recurrence override for that instance on the base event, and a
+   destroy as removing just that instance.
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 26]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   Clients MUST NOT send an update/destroy to both the base event and a
```

Comment:
Because detecting and returning an error is not always trivial, and it's unlikely a client would be doing both — you either build a full client that always handles the full event, or a basic client that always asks the server to do the expansions.